Refactor mkl fft planning to not use Tensor objects

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #47601 Use new FFT operators in stft
* #46912 Optimize torch.fft.fftn for real input with an r2c transform
* #46911 Improve torch.fft n-dimensional transforms
* **#46910 Refactor mkl fft planning to not use Tensor objects**

Differential Revision: [D25119656](https://our.internmc.facebook.com/intern/diff/D25119656)

peterbell10 committed:
Refactor mkl fft planning to not use Tensor objects

[ghstack-poisoned]

peterbell10 committed:
Update on "Refactor mkl fft planning to not use Tensor objects"




[ghstack-poisoned]

peterbell10 committed:
Update on "Refactor mkl fft planning to not use Tensor objects"




[ghstack-poisoned]

peterbell10 committed:
Update on "Refactor mkl fft planning to not use Tensor objects"




[ghstack-poisoned]

peterbell10 committed:
Update on "Refactor mkl fft planning to not use Tensor objects"




[ghstack-poisoned]

peterbell10 committed:
Update on "Refactor mkl fft planning to not use Tensor objects"




[ghstack-poisoned]

peterbell10 committed:
Update on "Refactor mkl fft planning to not use Tensor objects"




[ghstack-poisoned]

peterbell10 committed:
Update on "Refactor mkl fft planning to not use Tensor objects"




[ghstack-poisoned]

peterbell10 committed:
Update on "Refactor mkl fft planning to not use Tensor objects"




[ghstack-poisoned]

mruberry commented on aten/src/ATen/native/mkl/SpectralOps.cpp:
```diff
@@ -178,31 +178,109 @@ REGISTER_ARCH_DISPATCH(fft_fill_with_conjugate_symmetry_stub, DEFAULT, &_fft_fil
 REGISTER_AVX_DISPATCH(fft_fill_with_conjugate_symmetry_stub, &_fft_fill_with_conjugate_symmetry_cpu_)
 REGISTER_AVX2_DISPATCH(fft_fill_with_conjugate_symmetry_stub, &_fft_fill_with_conjugate_symmetry_cpu_)
 
+// For complex types, strides are in units of 2 * element_size(dtype)
```
Add a comment describing what the function actually does and its intended use

mruberry commented on aten/src/ATen/native/mkl/SpectralOps.cpp:
```diff
@@ -178,31 +178,109 @@ REGISTER_ARCH_DISPATCH(fft_fill_with_conjugate_symmetry_stub, DEFAULT, &_fft_fil
+  }
+
+  if (sizeof(MKL_LONG) < sizeof(int64_t)) {
+    TORCH_CHECK(signal_numel <= MKL_LONG_MAX,
```
Do we need to add a test case for this?

peterbell10 committed:
Update on "Refactor mkl fft planning to not use Tensor objects"




[ghstack-poisoned]

peterbell10 committed:
Update on "Refactor mkl fft planning to not use Tensor objects"




[ghstack-poisoned]

peterbell10 committed:
Update on "Refactor mkl fft planning to not use Tensor objects"




[ghstack-poisoned]

mruberry reviewed (approved):
Stamped.

peterbell10 committed:
Update on "Refactor mkl fft planning to not use Tensor objects"


Differential Revision: [D25119656](https://our.internmc.facebook.com/intern/diff/D25119656)

[ghstack-poisoned]
